pkg/server/serverpb/BUILD.bazel

```diff
@@ -4,40 +4,80 @@ load("@rules_proto//proto:defs.bzl", "proto_library")
 load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
 load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
 
+PROTO_LIBRARY_DEPS = [
+    "//pkg/build:build_proto",
+    "//pkg/clusterversion:clusterversion_proto",
+    "//pkg/config/zonepb:zonepb_proto",
+    "//pkg/gossip:gossip_proto",
+    "//pkg/jobs/jobspb:jobspb_proto",
+    "//pkg/kv/kvserver/kvserverpb:kvserverpb_proto",
+    "//pkg/kv/kvserver/liveness/livenesspb:livenesspb_proto",
+    "//pkg/roachpb:roachpb_proto",
+    "//pkg/server/diagnostics/diagnosticspb:diagnosticspb_proto",
+    "//pkg/server/status/statuspb:statuspb_proto",
+    "//pkg/sql/contentionpb:contentionpb_proto",
+    "//pkg/storage/enginepb:enginepb_proto",
+    "//pkg/ts/catalog:catalog_proto",
+    "//pkg/util:util_proto",
+    "//pkg/util/hlc:hlc_proto",
+    "//pkg/util/log/logpb:logpb_proto",
+    "//pkg/util/metric:metric_proto",
+    "@com_github_gogo_protobuf//gogoproto:gogo_proto",
+    "@com_google_protobuf//:timestamp_proto",
+    "@go_googleapis//google/api:annotations_proto",
+    "@io_etcd_go_etcd_raft_v3//raftpb:raftpb_proto",
+]
+
+# Define proto_library per each .proto file according to recommended code organization:
+# https://docs.bazel.build/versions/master/be/protocol-buffer.html#proto_library
 proto_library(
```

Comment:
So, this stuff is all auto-generated (if you check out the failing lint job, you'll see that it's yelling at you to run `make bazel-generate` -- but if you do, that'll trample all the changes you've made here).

[Gazelle](https://github.com/bazelbuild/bazel-gazelle) auto-generates the structure of the proto rules so that we don't have to manually update these dependencies ourselves. Any solution we arrive at for this will have to honor all the existing Gazelle-based machinery here. A couple high-level options (sorry, these are all hand-wavy, it's a complicated problem):

1. Exclude the `.proto` files in `pkg/server/serverpb` from being managed by Gazelle. This would mean that any change to the dependencies of any of the `.proto` files in here would have to be manually implemented in the `BUILD` files. As far as I'm concerned, this is a worst-case scenario -- I think we would have lots of angry engineers if we forced that on them.
2. Maybe you can adjust the `gazelle:proto_group` [directive](https://github.com/bazelbuild/bazel-gazelle#directives) in this directory to achieve the same thing, where we have one `proto_library` per `.proto` file? Worth playing with and seeing what that gets you.
3. Gazelle is supposed to support [extensions](https://github.com/bazelbuild/bazel-gazelle/blob/master/extend.rst) to accomplish this sort of thing. I am not sure how much work it would be to hack something together that accomplishes the same thing as this PR, but it is worth at least considering because it would be really nice to have one tool that's responsible for all our `BUILD` files in every language.

Reply:
@rickystewart , thanks for detailed suggestions! I would prefer to avoid manual changes in `.proto` files as well.
Taking into account that generated JS protobuf client can be minified to reduce file size, it might be okay just rely on generated `serverpb_proto` proto.